### ISSUE TITLE
Sharing variables across scripts does not work

### ISSUE THREAD

**Issue by ldorigo** [autokey/autokey#448]
## Classification:

Bug (I think?)

## Reproducibility:

Always

## Version

AutoKey version: 0.95.10

Used GUI (Gtk, Qt, or both): QT

Installed via: pacman (or was it AUR?) on archlinux.

Linux Distribution: Arch

## Summary

In this script: https://github.com/autokey/autokey/wiki/Scripting#key-wrapper it looks as if when you use engine.run_script("foo"), the script foo has access to variables defined in the script from which it's being called. When I try to do the same it doesn't work. If it's supposed to work that way, then it's a bug (it doesn't work and gives me a non defined error).

If it's not supposed to work that way, then this is 1. a documentation error (remove that script from the wiki) and 2. a feature request to allow sharing data/variables among scripts (store.set_global could be used, but changing run_script to accept an optional parameter "data" that gets passed to a variable in the target script would be nicer).

## Steps to Reproduce (if applicable)

Create two scripts, "source" and "target".

`source`:

```python
#Enter script code
test = " hello"
engine.run_script("target")
```

`target`:
```python
print(test)
```

## Expected Results

"test" gets printed to the console

## Actual Results

I get a "NameError: name 'test' is not defined".

**Comment by ldorigo** [comment]
I asked around in the gitter chat and it seems like using store.get/set_global_value is the way to go. So I think the solution here would simply be to remove that script from the wiki (or change it to reflect the proper way to do it)

**Comment by josephj11** [comment]
@ldorigo The wiki got reorganized, so your link no longer works. It's [here](https://github.com/autokey/autokey/wiki/Advanced-Scripts#key-wrapper) now.

@Elliria the above example script looks like it probably never worked. It needs to be fixed or removed (if you want to look at it).

As soon as the example is fixed or removed, this issue can be closed as completed.

In AutoKey, all normal variables are in the scope of the current script.

The engine API starts a new script with its own scope, so the variables from the calling script are not available in the called script.

As you discovered, using AutoKey stored local and global variables gets around this limitation. So does importing a module/functions from the Modules directory (but those can't contain any API calls without some extra work). 
So, bottom line is that this feature works as intended, but the example code appears to be wrong.

Adding data parameters to the engine API call is an interesting idea. That should be requested in a separate enhancement request issue. I don't know the code, so I'm not sure whether that would be easy or hard to implement.

@Elliria  You might want to glance at the code base and see if this looks like it would be easy to do. It seems a lot simpler than using the store API calls in this case.

**Comment by josephj11** [comment]
@ldorigo Just enjoyed that duct tape site you linked to!

**Comment by Elliria** [comment]
The [Key Wrapper](https://github.com/autokey/autokey/wiki/Advanced-Scripts#key-wrapper) example script code can be fixed by moving this section into the **wrap_selection** script:
```python
wrap = "<shift>+9"
wrap_close = "<shift>+0"
```
Would you like me to edit the example script?

By the way, thank you for mentioning making a change to the `engine.run_script()` API call, @ldorigo. When I took a quick look at the documentation for that API call, I noticed an errant curly-bracket in the example, [which has now been fixed](https://github.com/autokey/autokey/pull/885).

By the way, that documentation shows this as an example (with the errant curly-bracket removed):
```python
engine.run_script(description, 'foo', 'bar', foobar='foobar')
```
I tried messing around with that `foobar='foobar'` part with no success and the documentation doesn't give a working example. Do either of you know how to use such an option?

**Comment by josephj11** [comment]
So that's how you found the fried curlyque!

@sebastiansam55 might know something about that syntax, but the project itself must have a GitHub page, forum, or listserv somewhere.

**Comment by Elliria** [comment]
> So that's how you found the fried curlyque!

Yeah, isn't that wild? If not for the help of @ldorigo, who knows how long that curly-bracket would have stuck around. Now that I know how to edit the documentation, the plan is to go all through it and find anything that needs fixing.
 
> @sebastiansam55 might know something about that syntax, but the project itself must have a GitHub page, forum, or listserv somewhere.

Which project?

**Comment by josephj11** [comment]
The project that magically generates the new documentation pages.

**Comment by Elliria** [comment]
Ah, okay. Alas, they wouldn't be expected to know about how our documentation or how our code works. Their focus is on their own code.
